Correct Missing Information in Exterior Fenestration Shaded State Table

RKStrand opened the pull request:
Pull request overview
---------------------

- Fixes #10919 
- Fixes #11149 

### Description of the purpose of this PR

The Exterior Fenestration Shaded State Table produces information about windows when shading control is implemented by the user.  The problem is that when a window without a frame/divider is entered, nothing is shown in this table.  The fix here is to produce values in this table for windows whether or not the window has a frame/divider when shading control is requested by the user.  When a frame/divider exists, the table will report as before.  Without a frame divider, the table will now as a result of this fix show values relating to the U-Factor, SHGC, and visible transmittance for the glass only in the shaded state.  This includes a unit test to verify that this is reporting properly and fixes to the documentation (Input Output Reference) to clarify what will be reported.

### Expected Diffs ###
- New rows in the  Exterior Fenestration Shaded State Table for shaded constructions used without a frame/divider.
- New rows in the eio output for `FenestrationShadedState` for shaded constructions used without a frame/divider.
- Corrected values for Glass SHGC in the Exterior Fenestration Shaded State Table for all constructions with or without a frame/divider.

### Pull Request Author

 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer

<!-- This will not be exhaustively relevant to every PR. -->

 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Fix for Missing Report Information

This corrects the issue where the exterior fenestration shading state report is not producing output for valid window shading control input.  Before this report was only being populated for windows that had a frame or divider.  Now, shading control for windows without frames/dividers will also be listed in this predefined report.

RKStrand committed:
Merge branch 'develop' into 10919MissingInfoExtFenestrationShadeTable

RKStrand committed:
Unit Test of Fix for Missing Report Information

Creation of a unit test to fix the issue with a report that was missing.  Plus a typo correction in a comment of the fix.

RKStrand committed:
Exterior Fenestration Shaded State Docs Fix

Minor tweaks and corrections to the IO Ref relating to the fix made here.  This clarifies what will be reported and when.  Also a few minor edits were made for consistency.

mjwitte reviewed (commented):
Tested with WindowTests.idf. The table is populated now, but the U-Factor and SHGC values are identical to the unshaded construction. VT does vary.

RKStrand committed:
Corrections to Report from Reviewer

The values reported in the exterior fenestration shaded state table were not correct as they were using the unmodified glass values.  This has been corrected.  Modification to the unit test was also necessary.

RKStrand commented:
> Tested with WindowTests.idf. The table is populated now, but the U-Factor and SHGC values are identical to the unshaded construction. VT does vary.

@mjwitte I have fixed this in the latest commit that I just made.  I believe that the report should be correct now.  The unit test also had to be updated to reflect this change.

mjwitte commented:
Hmm. I made a test file that adds a skinny frame to all the windows in Windowtests. The new reported values for shaded glass SHGC and VT agree with the value with a frame (same for develop and this branch).
<img width="1033" height="297" alt="image" src="https://github.com/user-attachments/assets/1fd89646-4708-4c1c-9e38-0020bf8a7156" />
<img width="1067" height="330" alt="image" src="https://github.com/user-attachments/assets/354bb125-7889-4d85-95b5-df7f60f42428" />
Glass U-factors are slightly different with the frame for some reason, but that's not a problem for me.

But the shaded SHGC looks wrong (both with frame and without). You can look into that here, or make a new issue for that, or prove me wrong. It just seems impossible the the glass SHGC for Double Pane with Drapes is 0.746 when the unshaded SHGC is 0.764 and the assembly SHGC is 0.077.
<img width="1884" height="221" alt="image" src="https://github.com/user-attachments/assets/3b757f6b-ed4d-4723-afbf-3a376f9fb586" />
[WindowTests-AddFrames.idf.txt](https://github.com/user-attachments/files/21415931/WindowTests-AddFrames.idf.txt)

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 10919MissingInfoExtFenestrationShadeTable

mjwitte reviewed (approved):
@RKStrand This looks good. I've merged in develop and will let CI run one last time before merging.
Also posted a new issue #11149 about the shaded Glass SHGC values.

mjwitte commented:
@RKStrand I think I see the problem with the shaded SHGC values. I'm going to try to fix that in this same PR.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 10919MissingInfoExtFenestrationShadeTable

mjwitte committed:
Fix shaded glass shgc 11149

RKStrand commented:
@mjwitte Just trying to clarify, are you asking that reporting for unframed shaded windows also be sent to the EIO?

mjwitte commented:
> @mjwitte Just trying to clarify, are you asking that reporting for unframed shaded windows also be sent to the EIO?

Yes, please, if it's not too much trouble.

RKStrand commented:
@mjwitte Thanks for clarifying.  I'll look into it and get back to you.  I don't think it should be too much trouble but of course I could always be wrong here.

RKStrand committed:
Addition of reporting info to EIO

The information that was added to the HTML tabular report was not reporting to the EIO like it does for windows with frames/dividers.  This corrects this issue and now reports to the EIO.

mjwitte committed:
Avoid duplication in shaded fenestration reports

nrel-bot-2 commented:
@Myoldmopar @RKStrand @myoldmopar it has been 36 days since this pull request was last updated.

mitchute committed:
Merge branch 'develop' into 10919MissingInfoExtFenestrationShadeTable

mitchute committed:
fix build error
